src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -132,9 +132,12 @@ public interface IChainedHeaderTree
 
         /// <summary>
         /// Get the block and its chained header if it exists.
-        /// If the header is not in the tree returns <c>null</c>, the <see cref="ChainedHeaderBlock.Block"/> may also be null.
+        /// If the header is not in the tree <see cref="ChainedHeaderBlock"/> will be <c>null</c>, the <see cref="ChainedHeaderBlock.Block"/> may also be null.
         /// </summary>
-        /// <returns>The block and its chained header.</returns>
+        /// <remarks>
+        /// The block can be null when the block data has not yet been downloaded or if the block data has been persisted to the database and removed from the memory.
+        /// </remarks>
+        /// <returns>The block and its chained header (both the block and chained header can be null).</returns>
```

Comment:
Actually I don't think that chained header can be null. Either it is the whole result that is null, or the chained header has to value.

In any case, use `<c>` for null here and in the remarks and above in the summary

Reply:
Ah yea its the ChainedHeaderBlock that can be null.